tensorflow/lite/micro/kernels/ceva_bx1/fully_connected.cc

```diff
@@ -0,0 +1,348 @@
+
+  const TfLiteTensor* input = GetInput(context, node, kInputTensor);
+  const TfLiteTensor* filter = GetInput(context, node, kWeightsTensor);
+  const TfLiteTensor* bias = GetOptionalInputTensor(context, node, kBiasTensor);
+  TfLiteTensor* output = GetOutput(context, node, kOutputTensor);
+
+  TF_LITE_ENSURE_TYPES_EQ(context, input->type, output->type);
+  TF_LITE_ENSURE_MSG(context, input->type == filter->type,
+                     "Hybrid models are not supported on TFLite Micro.");
+
+  return CalculateOpData(context, params->activation, input->type, input,
+                         filter, bias, output, data);
+}
+
+TfLiteStatus EvalQuantizedInt8(TfLiteContext* context, TfLiteNode* node,
+                               const OpData& data,
+                               const TfLiteEvalTensor* input,
+                               const TfLiteEvalTensor* filter,
+                               const TfLiteEvalTensor* bias,
+                               TfLiteEvalTensor* output) {
+  tflite::FullyConnectedParams op_params;
+  op_params.input_offset = -data.input_zero_point;
+  op_params.weights_offset = -data.filter_zero_point;
+  op_params.output_offset = data.output_zero_point;
+  op_params.output_multiplier = data.output_multiplier;
+  // TODO(b/138810107): Figure out whether output shift should be inverted
+  op_params.output_shift = -data.output_shift;
+  op_params.quantized_activation_min = data.output_activation_min;
+  op_params.quantized_activation_max = data.output_activation_max;
+
+#ifdef ORIGINAL_IMPLEMENTATION
+
+  reference_integer_ops::FullyConnected(
+      op_params, tflite::micro::GetTensorShape(input),
+      tflite::micro::GetTensorData<int8_t>(input),
+      tflite::micro::GetTensorShape(filter),
+      tflite::micro::GetTensorData<int8_t>(filter),
+      tflite::micro::GetTensorShape(bias),
+      tflite::micro::GetTensorData<int32_t>(bias),
+      tflite::micro::GetTensorShape(output),
+      tflite::micro::GetTensorData<int8_t>(output));
+#else  // ORIGINAL_IMPLEMENTATION
+  int input_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(input).DimensionsCount();
+  int weights_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(filter).DimensionsCount();
+  int* weights_shape_DimsData =
+      (int*)tflite::micro::GetTensorShape(filter).DimsData();
+  int bias_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(bias).DimensionsCount();
+  int output_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(output).DimensionsCount();
+  int* output_shape_DimsData =
+      (int*)tflite::micro::GetTensorShape(output).DimsData();
+
+  void* params = (void*)&op_params;
+  int8_t* inputp = (int8_t*)tflite::micro::GetTensorData<int8_t>(input);
+  int8_t* filterp = (int8_t*)tflite::micro::GetTensorData<int8_t>(filter);
+  int32_t* biasp = (int32_t*)tflite::micro::GetTensorData<int32_t>(bias);
+  int8_t* outputp = (int8_t*)tflite::micro::GetTensorData<int8_t>(output);

```

Comment:
prefer C++ style casts, and static_cast whenever possible.

https://google.github.io/styleguide/cppguide.html#Casting